### ISSUE TITLE
Allow functions to return multiple cardinality

### ISSUE THREAD

**Issue by minesh-s-patel** [REGnosys/rosetta-dsl#135]
Functions are constrained so they can only result is an output with a singular cardinality. This should be enhanced so a function results with a milti cardinality.

Code generators should be updated to support this.
Type safety on the resulting multi card should be respected.

**Comment by dhuebner** [comment]
Depends on #71

**Comment by SimonCockx** [comment]
At this time, the Rosetta validator accepts functions with multi cardinality, but the syntax of Rosetta does not allow to index the output parameter, e.g. this is disallowed.
```
func MultiOutputTest:
	output: result int (2..2)
	assign-output result[0]:
		0
```
The validator also wrongly accepts something like this:
```
type Container:
	val int (1..1)
func MultiPathTest:
	output: result Container (2..2)
	assign-output result->val:
		1
```
although the Java generator generates uncompilable code.

**Comment by SimonCockx** [comment]
Functions returning values with a multiple cardinality are supported. For the validation issue in my previous comment, I'm raising a separate issue: https://github.com/REGnosys/rosetta-dsl/issues/391.
